### PR TITLE
Change default inSubQueryThreshold

### PR DESCRIPTION
**Pull request by adarshsanjeev**
Currently, the default inSubQueryThreshold is Integer.MAX_VALUE. This was chosen at the time to maintain the then current behavior. However, this value is too high for most cases, and significant slowdown has been observed for large number of values, compared to joins.

- Change the default value of inSubQueryThreshold to 20, causing in filters with more elements to be planned as joins instead of OR filters.

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by abhishekagarwal87 — APPROVED** [review]

File: sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java [hunk 1]
```diff
@@ -5269,44 +5267,6 @@ public void testSqlIsNullToInFilter()
     );
   }
 
-  @Test
-  public void testInFilterWith23Elements()
```
Comment (kgyrtkirk) [inline sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java:5273]: shouldn't this test be kept?
I think if it gets removed the subq transformed codepath is not covered by the other tests

Comment (adarshsanjeev) [reply sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java:5273]: `org.apache.druid.sql.calcite.CalciteQueryTest#testPlanWithInFilterLessThanInSubQueryThreshold` and `org.apache.druid.sql.calcite.CalciteJoinQueryTest#testPlanWithInFilterMoreThanInSubQueryThreshold` should cover both sides of this. 
The comment in this ticket implies that it was for a particular issue when druid did not support joins. Is there a reason to keep it?

File: docs/querying/query-context.md [hunk 2]
```diff
@@ -66,7 +66,7 @@ Unless otherwise noted, the following parameters apply to all query types.
...
-|`inSubQueryThreshold`|`2147483647`| Threshold for minimum number of values in an IN clause to convert the query to a JOIN operation on an inlined table rather than a predicate. A threshold of 0 forces usage of an inline table in all cases; a threshold of [Integer.MAX_VALUE] forces usage of OR in all cases. |
+|`inSubQueryThreshold`|`20`| Threshold for minimum number of values in an IN clause to convert the query to a JOIN operation on an inlined table rather than a predicate. A threshold of 0 forces usage of an inline table in all cases; a threshold of [Integer.MAX_VALUE] forces usage of OR in all cases. |
```
Comment (kgyrtkirk) [inline docs/querying/query-context.md:69]: note: I'm not sure but isn't the wording here suggests that setting it to `Integer.MAX_VALUE` will create `OR`-s from everything?

**Review by kgyrtkirk — APPROVED** [review]

File: docs/querying/query-context.md (hunk 2 above)
Comment (adarshsanjeev) [reply docs/querying/query-context.md:69]: That should be what happens if the number of entries is less than the threshold.

**Pull request merged**